### PR TITLE
resource/aws_servicecatalog_provisioned_product: Return clearer error message on failure

### PR DESCRIPTION
**Pull request by gdavison**
### Description

When Provisioned Product fails to apply, it either returns a status of `ERROR` on creation on `TAINTED` on update when it can roll back to a previous version.

Return the error information to the practitioner so that they know it has failed. Also return the error as a top-level error instead of buried in an "unexpected state" error.

### Relations

Supersedes #30522

### Output from Acceptance Testing

```
$ make testacc PKG=servicecatalog TESTS=TestAccServiceCatalogProvisionedProduct_

--- PASS: TestAccServiceCatalogProvisionedProduct_taintedCreate (60.70s)
--- PASS: TestAccServiceCatalogProvisionedProduct_basic (141.19s)
--- PASS: TestAccServiceCatalogProvisionedProduct_disappears (143.46s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags (226.40s)
--- PASS: TestAccServiceCatalogProvisionedProduct_update (233.87s)
--- PASS: TestAccServiceCatalogProvisionedProduct_computedOutputs (249.37s)
--- PASS: TestAccServiceCatalogProvisionedProduct_taintedUpdate (269.72s)
```

### CODE REVIEW

File: .changelog/30588.txt
Comment (ewbankkit) [inline .changelog/30588.txt:1]: Shouldn't this be `30663.txt`?

Comment (gdavison) [reply .changelog/30588.txt:1]: Yes, not sure where I got that number…

File: .changelog/30588.txt [hunk 1]
```diff
@@ -1,3 +0,0 @@
-```release-note:bug
-resource/aws_api_gateway_stage: `cache_cluster_size` is not updated if `cache_cluster_size` is `false`
-```
```
Comment (jar-b) [inline .changelog/30588.txt:3]: Was this removal supposed to be 30522?

**Review by jar-b — APPROVED** [review]
LGTM 

```console
$ make testacc PKG=servicecatalog TESTS=TestAccServiceCatalogProvisionedProduct_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/servicecatalog/... -v -count 1 -parallel 20 -run='TestAccServiceCatalogProvisionedProduct_'  -timeout 180m
=== RUN   TestAccServiceCatalogProvisionedProduct_basic
=== PAUSE TestAccServiceCatalogProvisionedProduct_basic
=== RUN   TestAccServiceCatalogProvisionedProduct_update
=== PAUSE TestAccServiceCatalogProvisionedProduct_update
=== RUN   TestAccServiceCatalogProvisionedProduct_computedOutputs
=== PAUSE TestAccServiceCatalogProvisionedProduct_computedOutputs
=== RUN   TestAccServiceCatalogProvisionedProduct_disappears
=== PAUSE TestAccServiceCatalogProvisionedProduct_disappears
=== RUN   TestAccServiceCatalogProvisionedProduct_tags
=== PAUSE TestAccServiceCatalogProvisionedProduct_tags
=== RUN   TestAccServiceCatalogProvisionedProduct_errorOnCreate
=== PAUSE TestAccServiceCatalogProvisionedProduct_errorOnCreate
=== RUN   TestAccServiceCatalogProvisionedProduct_errorOnUpdate
=== PAUSE TestAccServiceCatalogProvisionedProduct_errorOnUpdate
=== CONT  TestAccServiceCatalogProvisionedProduct_basic
=== CONT  TestAccServiceCatalogProvisionedProduct_tags
=== CONT  TestAccServiceCatalogProvisionedProduct_computedOutputs
=== CONT  TestAccServiceCatalogProvisionedProduct_errorOnUpdate
=== CONT  TestAccServiceCatalogProvisionedProduct_errorOnCreate
=== CONT  TestAccServiceCatalogProvisionedProduct_disappears
=== CONT  TestAccServiceCatalogProvisionedProduct_update
--- PASS: TestAccServiceCatalogProvisionedProduct_errorOnCreate (53.83s)
--- PASS: TestAccServiceCatalogProvisionedProduct_disappears (133.85s)
--- PASS: TestAccServiceCatalogProvisionedProduct_basic (134.79s)
--- PASS: TestAccServiceCatalogProvisionedProduct_tags (229.65s)
--- PASS: TestAccServiceCatalogProvisionedProduct_update (232.50s)
--- PASS: TestAccServiceCatalogProvisionedProduct_computedOutputs (247.29s)
--- PASS: TestAccServiceCatalogProvisionedProduct_errorOnUpdate (270.94s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/servicecatalog     277.528s
```

File: .changelog/30588.txt (hunk 1 above)
Comment (gdavison) [reply .changelog/30588.txt:3]: Oh my goodness! Good catch

**Pull request merged**